optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/forager/finalist/HighestScoreFinalistPodium.java

```diff
@@ -47,7 +48,7 @@ public void addMove(LocalSearchMoveScope moveScope) {
         }
         Score moveScore = moveScope.getScore();
         int scoreComparison = doComparison(moveScore);
-        if (scoreComparison > 0) {
+        if (scoreComparison > 0 || finalistList.size() > finalistMaxSize) {
```

Comment:
Fixing this slightly different:
We can't accept a move just because some sort of hard threshold has been reached. That hard number is both too low and to big.
But we can limit the growth of the finalistList. Doing that.